src/vanilla.ts

```diff
@@ -63,10 +67,11 @@ export default function create<TState extends State>(
       try {
         const newStateSlice = selector(state)
         if (!equalityFn(currentSlice, newStateSlice)) {
-          listener((currentSlice = newStateSlice))
+          const previousSlice = currentSlice
+          listener((currentSlice = newStateSlice), previousSlice)
         }
-      } catch (error) {
-        listener(null, error)
+      } catch {
+        listener(null, null)
```

Comment:
I don't think we need to call listeners after an error in vanilla. We needed to do it to force React to re-render and that logic still exists in index.ts. The listeners added in index.ts are also wrapped in a try-catch so they won't throw.

My suggestion: remove the listener call in the catch.

Reply:
Done :)

Reply:
Yeah, I should have clarified more about the context. Check #208 and we think it's better not to try-catch.